apps/rest/rest.app.js

```diff
@@ -0,0 +1,322 @@
+  g.setColor(SET_COLOR);
+  roundRect(q2.x, q2.y, q2.x2, q2.y2, 20);
+  g.setColor(GREEN_COLOR_MUTED);
+  roundRect(q4.x, q4.y, q4.x2, q4.y2, 20);
+  g.setColor(-1);
+  g.setFont(BIG_FONT).setFontAlign(0,0).drawString("SET", c2.x, c2.y-25)
+  g.setFont(HUGE_FONT).setFontAlign(0,0).drawString("#"+currentSet, c2.x, c2.y)
+  g.setFont(BIG_FONT).setFontAlign(0,0).drawString("PUSH >\nBUTTON\nWHEN\nDONE", c4.x, c4.y)
+}
+
+function circlePoints (cx, cy, r, points) {
+  let circlePoints = [];
+  for (let i=0; i<points; i++) {
+    circlePoints.push(-Math.sin(i/points*Math.PI*2) * r + cx);
+    circlePoints.push(Math.cos(i/points*Math.PI*2) * r + cy);
+  }
+  return circlePoints;
+}
+
+const smallQ3Circle = [c2.x, c2.y].concat(circlePoints(c2.x, c2.y, ((q2.y - q2.y2)/2) + 10, 60));
+
+
+function drawRest() {
+  const start = Date.now();
+  const secondsRemaining = Math.max(0, ((restUntil - Date.now()) / 1000) | 0);
+
+  g.setColor(REST_COLOR);
+  roundRect(q1.x, q1.y, q1.x2, q1.y2, 20);
+  g.setColor(-1).setFont(HUGE_FONT).setFontAlign(0,0).drawString(secondsRemaining, c1.x, c1.y)
+
+  g.setColor(SET_COLOR_MUTED);
+  roundRect(q2.x, q2.y, q2.x2, q2.y2, 20);
+  const factor = 1 - secondsRemaining / restSeconds;
+  const circleParts = (((factor * smallQ3Circle.length) | 0) >> 1) << 1
+  const poly = smallQ3Circle.slice(0, circleParts + 2)
+  g.setColor(SET_COLOR);
+  g.fillPoly(poly);
+
+  g.setColor(GREY_COLOR);
+  roundRect(q3.x, q3.y, q3.x2, q3.y2, 20);
+  g.setColor(-1).setFont(BIG_FONT).setFontAlign(0,0).drawString("REST", c3.x, c3.y)
+
+  g.setColor(0);
+  g.setFont("6x8").drawString("Push button\nto skip ->", c4.x, c4.y);
+
+  if (secondsRemaining > 0) {
+    if (secondsRemaining < 5) {
+      if (secondsRemaining > 1) {
+        Bangle.buzz(100);
+      } else {
+        Bangle.buzz(1000);
+      }
+    }
+    const renderTime = Date.now() - start;
+    setTimeout(redrawApp, Math.max(10, 1000 - renderTime));
+  } else {
+    currentSet += 1;
+    if (currentSet > setsCount) {
+      currentSet = 1;
+      setMode(MAIN_MENU);
+    } else {
+      setMode(EXERCISE);
+    }
+    redrawApp();
+  }
+}
+
+function drawDoIt() {
+  g.setBgColor('#00FF00').clear();
+  g.drawImage(getImg(), 44, 44);
+  g.setFont(BIG_FONT)
+  g.setColor(0);
+  setTimeout(() => {
+    g.setFontAlign(0, 0)
+    g.drawString('just      ', R.x2/2, 20);
+    Bangle.buzz(150, 0.5);
+  }, 200);
+  setTimeout(() => {
+    g.drawImage(getImg(), 22, 44, {scale: 1.5});
+    g.drawString('     DO   ', R.x2/2, 20);
+    Bangle.buzz(200);
+  }, 1000);
+  setTimeout(() => {
+    g.drawString('        IT', R.x2/2, 20);
+    Bangle.buzz(200);
+  },  1400);
+  setTimeout(() => {
+    setMode(MAIN_MENU);
+    redrawApp();
+  }, 2000);
+}
+
+const MAIN_MENU = 'MAIN_MENU';
+const SET_REST = 'SET_REST';
+const SET_SETS = 'SET_SETS';
+const EXERCISE = 'EXERCISE';
+const REST = 'REST';
+const DOIT = 'DOIT';
+
+let mode = MAIN_MENU;
+let lastRender = MAIN_MENU;
```

Comment:
We get a lint warning here. `'lastRender' is assigned a value but never used` . Please either remove the line or comment it out.